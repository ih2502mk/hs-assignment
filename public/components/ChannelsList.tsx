import React, { FC } from "react";
import Channel from "../../shared/Channel";

interface ChannelsListProps {
    channels: Channel[];
    onSelectChannel: (id: Channel["id"]) => void;
}

export const ChannelsList: FC<ChannelsListProps> = ({
    channels,
    onSelectChannel
}) => {
    return (
        <ul>
            {channels.map(({ name, id }) => (
                <li key={id}>
                    <button onClick={() => onSelectChannel(id)}>{name}</button>
                </li>
            ))}
        </ul>
    );
};
