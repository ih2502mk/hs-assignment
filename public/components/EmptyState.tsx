import React, { FC, ReactNode } from "react";
import styled from "styled-components";

const Wrapper = styled.div`
    color: lightgray;
    text-align: center;
    width: 100%;
`;

export const EmptyState: FC<ReactNode> = ({ children }) => {
    return <Wrapper>{children}</Wrapper>;
};
