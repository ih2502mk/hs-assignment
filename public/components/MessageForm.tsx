import React, { FC } from "react";
import Message from "../../shared/Message";

interface MessageFormProps {
    message: Message;
}

export const MessageForm: FC<MessageFormProps> = ({ message }) => {
    return <></>;
};
