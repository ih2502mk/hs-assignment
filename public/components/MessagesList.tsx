import React, { FC } from "react";
import Message from "../../shared/Message";

interface MessagesListProps {
    messages: Message[];
}

export const MessagesList: FC<MessagesListProps> = ({ messages }) => {
    return (
        <ul>
            {messages.map(({ content, id }) => (
                <li key={id}>{content}</li>
            ))}
        </ul>
    );
};
