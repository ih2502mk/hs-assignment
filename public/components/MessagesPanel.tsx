import React, { FC } from "react";
import Message from "../../shared/Message";
import Channel from "../../shared/Channel";
import { MessagesList } from "./MessagesList";
import { MessageForm } from "./MessageForm";

interface MessagesPanelProps {
    channelId: Channel["id"];
}

export const MessagesPanel: FC<MessagesPanelProps> = ({ channelId }) => {
    const messages: Message[] = [];

    const editedMessage: Message = {
        content: "",
        id: null,
        channelId,
        created: new Date(),
        updated: new Date()
    };

    return (
        <>
            <MessagesList messages={messages} />
            <MessageForm message={editedMessage} />
        </>
    );
};
