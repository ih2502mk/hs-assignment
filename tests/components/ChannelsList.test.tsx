import React from "react";
import renderer from "react-test-renderer";
import { ChannelsList } from "../../public/components/ChannelsList";

describe("snapshots", () => {
    it("renders a list of channels", () => {
        const component = renderer
            .create(<ChannelsList channels={[]} onSelectChannel={() => {}} />)
            .toJSON();

        expect(component).toMatchSnapshot();
    });
});
