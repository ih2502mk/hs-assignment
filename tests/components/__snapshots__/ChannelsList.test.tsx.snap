// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders a list of channels 1`] = `<ul />`;
